File: rails/app/data/deck.rb
# A class that contains 52 cards (CardState).  It shuffles them initially
class Deck
  attr_accessor :cards

  SHUFFLE_FACTOR = 1000

  # @type [Array]
  @cards

  def initialize
    super
    @cards = Array.new
    add_default_cards
    shuffle
  end

  def add_default_cards
    @cards.concat(Card.all.map { |c| CardState.new(c.suit.short_form, c.card_value.short_form, c.card_value.value) })
  end

  def shuffle
    last_index = @cards.length - 1
    (0..last_index).each { |i| swap(@cards, i, rand(last_index)) }
    last_index.downto(0) { |i| swap(@cards, i, rand(last_index)) }
    SHUFFLE_FACTOR.times { |i| swap(@cards, rand(last_index), rand(last_index)) }
  end

  def swap(cards, item1, item2)
    tmp = cards[item1]
    cards[item1] = cards[item2]
    cards[item2] = tmp
  end
end

File: django/blackjack/store/deck.py

from blackjack.models import Card as CardModel
import random

from blackjack.state.card import Card

class Deck:
    cards: list[Card]

    def __init__(self) -> None:
        self.cards = list()
        self.addDefaultCards()
        self.shuffle()

    def addDefaultCards(self):
        self.cards.extend(map(lambda c: Card(c.suit.short_form, c.card.short_form, c.card.value), CardModel.objects.all()))

    def shuffle(self):
        SHUFFLE_FACTOR = 1000    
        deck_size = len(self.cards)
        for i in range(0, deck_size):
            self.swap(self.cards, i, random.randint(0, deck_size - 1))
        for i in range(deck_size - 1, -1, -1):
            self.swap(self.cards, i, random.randint(0, deck_size - 1))
        for i in range(0, SHUFFLE_FACTOR):
            self.swap(self.cards, random.randint(0, deck_size - 1), random.randint(0, deck_size - 1))

    def swap(self, cards: list[Card], item1: int, item2: int):
        tmp = cards[item1]
        cards[item1] = cards[item2]
        cards[item2] = tmp